topdown/reachable.go

```diff
@@ -72,43 +72,59 @@ func builtinReachable(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.
 	return iter(ast.NewTerm(reached))
 }
 
-// pathBuilder is called recursively to build an array of paths that are reachable from the root
-func pathBuilder(graph ast.Object, root *ast.Term, path []*ast.Term, paths []*ast.Term, reached ast.Set) []*ast.Term {
+// pathBuilder is called recursively to build a Set of paths that are reachable from the root
+func pathBuilder(graph ast.Object, root *ast.Term, path []*ast.Term, edgeRslt ast.Set, reached ast.Set) ast.Set {
+	paths:= []*ast.Term{}
+
 	if edges := graph.Get(root); edges != nil {
 		path = append(path, root)
 
 		if numberOfEdges(edges) >= 1 {
+
 			foreachVertex(edges, func(neighbor *ast.Term) {
+
 				if reached.Contains(neighbor) {
 					// If we've already reached this node, return current path (avoid infinite recursion)
 					paths = append(paths, path...)
+					edgeRslt.Add(ast.ArrayTerm(paths...))
 				} else {
 					reached.Add(root)
-					paths = pathBuilder(graph, neighbor, path, paths, reached)
+					pathBuilder(graph, neighbor, path, edgeRslt, reached)
+
 				}
+
+
 			})
+
 		} else {
 			paths = append(paths, path...)
+			edgeRslt.Add(ast.ArrayTerm(paths...))
+
+
 		}
 	} else {
 		// Node is nonexistent (not in graph). Commit the current path (without adding this root)
 		paths = append(paths, path...)
+		edgeRslt.Add(ast.ArrayTerm(paths...))
+
+
 	}
+	return edgeRslt
 
-	return paths
 }
 
-func builtinReachablePaths(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
-	// Return an error if the first argument is not an object.
-	graph, err := builtins.ObjectOperand(operands[0].Value, 1)
+func builtinReachablePaths(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
+	var traceResult = ast.NewSet()
+	// Error on wrong types for args.
+	graph, err := builtins.ObjectOperand(args[0].Value, 1)
 	if err != nil {
 		return err
 	}
 
 	// This is a queue that holds all nodes we still need to visit.  It is
 	// initialised to the initial set of nodes we start out with.
 	var queue []*ast.Term
-	switch initial := operands[1].Value.(type) {
+	switch initial := args[1].Value.(type) {
```

Comment:
[nit] It's a matter of preference, but I believe most of the time, this is called `operands`. Let's keep it like that, please.